design/open/design-datasets.md

```diff
@@ -0,0 +1,211 @@
+## Existing
+
+We have these forms of table definitions.  I order them here roughly by
+chronological order of initial implementation, mixed with mutual
+dependencies.
+
+### `lakectl metastore`
+
+This command supports *Hive* and *Glue* metastores.
+
+A Hive metastore is stored off lakeFS on a Hive Metastore Server (HMS).
+`lakectl` allows updating the table stored on the HMS to match branch
+states, and can drive HMS merges and copies.  All metadata is stored on the
+HMS; the `lakectl metastore` command calls out to a HMS URI that is supplied
+by configuration or by flags.
+
+A Glue or similar metastore is stored off lakeFS.  lakeFS exports "symlinks"
+objects that give it direct access to S3.  All important table metadata is
+still stored on an external Glue catalog.
+
+### Delta Lake and Delta-diff
+
+Given a path, Delta describes its own schema, so no further catalog is
+necessary.  The current lakeFS Delta Lake implementation is unaware of the
+list of tables stored in the repository.
+
+Delta diff has no explicit concept of tables because it needs none.  The web
+UI activates Delta diff (only) when a folder contains a `_delta_log/`
+subfolder.
+
+It will be possible to list all subfolders holding Delta Lake folders by
+listing with separator `_delta_log/`.  (A simple API change could add a
+"prefixes-only" mode to the lakeFS list-objects API, if needed.)
+
+### Iceberg Catalog
+
+lakeFS Iceberg support uses a catalog to locate Iceberg files.  This allows
+it to translate schema to refs.  But the lakeFS Iceberg catalog wraps a
+Spark catalog; it does not store any tables itself.  Currently only
+SparkCatalog is supported, but it is not clear what _other_ catalogs would
```

Comment:
```suggestion
Hadoop catalog; it does not store any tables itself.  Currently only
HadoopCatalog is supported, but it is not clear what _other_ catalogs would
```
Also I wouldn't say that the Hadoop catalog is supported per se, rather it's `LakeFSCatalog` which behaves like a Hadoop catalog